diff --git a/deps/RadioHead/RHGenericDriver.h b/deps/RadioHead/RHGenericDriver.h
index b36533a..ea8c401 100644
--- a/deps/RadioHead/RHGenericDriver.h
+++ b/deps/RadioHead/RHGenericDriver.h
@@ -58,6 +58,11 @@ public:
     /// Constructor
     RHGenericDriver();

+    /// Virtual destructor to prevent compiler warning "warning: deleting object
+    /// of polymorphic class type ‘RH_Serial’ which has non-virtual destructor
+    /// might cause undefined behaviour [-Wdelete-non-virtual-dtor]"
+    virtual ~RHGenericDriver(){};
+
     /// Initialise the Driver transport hardware and software.
     /// Make sure the Driver is properly configured before calling init().
     /// \return true if initialisation succeeded.
